### ISSUE TITLE
Delta filter not working under Kalman-combinator

### ISSUE THREAD

**Issue by Aljosha89** [esphome/issues#4436]
### The problem

After update 2023.4.0 is no longer possible to use delta filter under kalman-combinator based sensor:

![image](https://user-images.githubusercontent.com/96951985/233442890-ff98c64c-6e0d-4adf-9d8c-500c29b2e874.png)

Tried to put value in integer or percentage, but there is no change on the problem. Same code under sensor still works. Before update code was working fine.

### Which version of ESPHome has the issue?

2023.4.0

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2023.4.4

### What platform are you using?

ESP32

### Board

esp32dev

### Component causing the issue

kalman_combinator

### Example YAML snippet

```yaml
- platform: kalman_combinator
  name: "Kalman level"
  id: kalman_level_tank
  unit_of_measurement: "%"
  process_std_dev: 0.020
  sources:
    - source: level_tank
      error: 0.850
    - source: level_tank_bis
      error: 0.850
  filters:
  - filter_out: nan
  - delta: 0.2
```

### Anything in the logs that might be useful for us?

```txt
Delta filter requires a positive number or percentage value.
```

### Additional information

![Schermata del 2023-04-20 19-27-44](https://user-images.githubusercontent.com/96951985/233443704-44bd6e57-b2f7-43d1-af1e-251e8e80eff7.png)

**Comment by ssieb** [comment]
Try compiling it and see what error message that gives.

**Comment by Aljosha89** [comment]
Here we are:

![image](https://user-images.githubusercontent.com/96951985/233445325-651eebb0-13af-4960-b950-d5b8b5e6bc15.png)

**Comment by ssieb** [comment]
Ok, I can reproduce it, but that's very strange!

**Comment by ssieb** [comment]
Actually, there is a clue in the compile logs.  It's doing something weird with the parameter, converting to a list and adding a `False`.

**Comment by jesserockz** [comment]
I have made a PR to fix the delta filter in https://github.com/esphome/esphome/pull/4723.

You should be able to test with this YAML added to your config:

```yaml
external_components:
  - source: github://pr#4723
    components: sensor
```

**Comment by Aljosha89** [comment]
Yes, now I can compile it and it works properly.

Waiting for definitive fix, thanks


________________________________
Da: Jesse Hills ***@***.***>
Inviato: venerdì 21 aprile 2023, 10:26
A: esphome/issues ***@***.***>
Cc: Aljosha89 ***@***.***>; Author ***@***.***>
Oggetto: Re: [esphome/issues] Delta filter not working under Kalman-combinator (Issue #4436)


I have made a PR to fix the delta filter in esphome/esphome#4723<https://github.com/esphome/esphome/pull/4723>.

You should be able to test with this YAML added to your config:

external_components:
  - source: github://pr#4723
    components: sensor
